support/provider.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package support
+
+type ClusterProvider string
+
+const (
+	Kind ClusterProvider = "Kind"
+)
+
+var ProviderRegistry map[ClusterProvider]E2EClusterProvider
+
+func GetClusterProvider(provider ClusterProvider) E2EClusterProvider {
+	return ProviderRegistry[provider]
+}
+
+func RegisterProvider(name ClusterProvider, provider E2EClusterProvider) {
+	ProviderRegistry[name] = provider
```

Comment:
This should be protected in with a sync.Mutex.

Reply:
Ack